cmd/launcher/internal/release_manager.go

```diff
@@ -125,18 +130,41 @@ func (rm *ReleaseManager) DownloadRelease(version string, progressCallback func(
 		rm.GitHubOwner, rm.GitHubRepo, version, version)
 
 	checksumPath := filepath.Join(rm.BinaryPath, "checksums.txt")
-	if err := rm.downloadFile(checksumURL, checksumPath, nil); err != nil {
-		return fmt.Errorf("failed to download checksums: %w", err)
-	}
 
-	// Verify the checksum
-	if err := rm.VerifyChecksum(localPath, checksumPath, binaryName); err != nil {
-		return fmt.Errorf("checksum verification failed: %w", err)
+	// Try to download checksums with retry logic
+	downloadErr := rm.downloadFile(checksumURL, checksumPath, nil)
+
+	// If download fails, check for manually placed checksum file
+	if downloadErr != nil {
+		log.Printf("Warning: failed to download checksums: %v", downloadErr)
+
+		// Check if user has manually placed a checksums file
+		manualChecksumPath := filepath.Join(rm.ChecksumsPath, fmt.Sprintf("checksums-%s.txt", version))
```

Comment:
would actually make sense here to check if first there is already a checksum file and honor that, and skipping download entirely in such case